components/dashboard/src/service/service.ts

```diff
@@ -49,22 +38,38 @@ function createGitpodService<C extends GitpodClient, S extends GitpodServer>() {
             }
         });
 
-        if (_websocket && "reconnect" in _websocket) {
-            reconnect = () => { (_websocket as any).reconnect() };
-        }
     // }
     const service = new GitpodServiceImpl<C, S>(proxy);
-    (service as any).reconnect = reconnect;
+    (service as any).reconnect = () => {
+        const ws = (webSocket as any);
+        if (typeof ws.reconnect === "function") {
+            ws.reconnect();
+        } else {
+            console.log("WebSocket reconnect not possible.");
+        }
+    };
     return service;
 }
 
 declare global {
-    type GitpodService = ReturnType<typeof createGitpodService> & { reconnect: () => void }
-    interface Window { gitpodService?: GitpodService; }
+    interface Window {
+        _gp: {
+            gitpodService: GitpodService;
+        }
+    }
 }
 
-// reuse existing service object if present
-let gitpodService: GitpodService = window.gitpodService || (window.gitpodService = createGitpodService() as GitpodService);
-// let gitpodService: GitpodService = gitpodServiceMock;
+window._gp = {
```

Comment:
the global window object should be internal. 
Clients should use a function that is exposed from here.
e.g. `getGitpodService()`.

Reply:
got it!